inluxclient/query.go

```diff
@@ -98,12 +98,15 @@ func (r *QueryResultReader) errorSection() error {
 			}
 		}
 		row := r.Row()
-		message, ok1 := row[0].(string)
-		reference, ok2 := row[1].(int64)
-		if !ok1 || !ok2 {
+		if row[0] == "" {
+			return errors.New("no row found in error section")
+		}
+		var e QueryError
+		if err := r.Decode(&e); err != nil {
+
 			return fmt.Errorf("unexpected column types (%T, %T) in error section", row[0], row[1])
```

Comment:
I'd be inclined to keep the decode error in there.
```suggestion
			return fmt.Errorf("cannot decode row in error section: %v", err)
```